### ISSUE TITLE
Buff auras (icons) are shown when they are not supposed to

### ISSUE THREAD

**Issue by Weeeqt** [WeakAuras/WeakAuras2#2249]
## Describe the bug
I made WA to track auras on my current target with Aura-Target-Buff trigger with show on auras found. In situation where a buff wears off from some unit and you target this unit at this exact moment when buff wears off - sometimes the buff icon persists and stays forever unless I change the target or this unit receives/refreshes new buffs/debuffs.

### To Reproduce

1. Make a simple Icon WA with Aura trigger for Unit: Target, Aura type: Buff, Debuff Type: Magic, Show On: Aura(s) Found. Enable Auro-Clone (Show All Matches).
2. Find a unit with some short magic buff or cast some magic buff on a unit yourself. Do not target this unit.
3. When that buff is about to end (like less than 0,1 sec left) - target that unit.
4. If the timing is correct you will see the problem.

**Screenshots**
I have a clip from twitch stream when this happened. I was also able to reproduce it later manually, but not always because the target must be changed at the same exact moment as when aura wears off and it's a bit random/hard to time it correctly.
In the clip enemy holy paladin uses Divine Favor and we can see it above his nameplate. I change the target to paladin too late, right when Divine Favor wears off but it still persists above my target - you can see the glowing icon there:
https://clips.twitch.tv/AnnoyingFreezingLatteDoubleRainbow

**Did you try having WeakAuras as the only enabled addon and everything else (especially something like ElvUI) disabled?**
No

**Which version of WeakAuras are you using?**
2.17.4

**Are you on World of Warcraft Classic or Retail?**
- [] Classic
- [X] Retail

### Additional Info
I removed all the unnecessary effects from the aura, such as glow etc and made sure the bug still persists with it too. It is really just a basic aura trigger for target magic-only buffs.

!vF1xZPnoq8pl3d3mxV7AgGKM27LBgGcT0Hysronj9MAqylmQ1irLKdH8G)SF7UYgCiqU(qF5MKXrA1Qv7F)TBIAgfgXSrSwN081NCweBzelb)fOYnXdfQu3IOGtpVreBM2KimD4XFlXOxfX6KjF4bUjPyAOwN5KaP7hnFUv4Ica23uTUzdyxQrMeUzLiIn(TvIQRotBGxkOjEHGw03tPVNHeDGMTIhlWJy8yNuRSK664ghSaoxQKLRMdlTlO1WoNrMMkmwV0d3saxMiMLpFUxzEFVHx2)QHW50wEUH3cFvrw2GeRxY28zI7ekhdUL8(i2KUTzHtyHThdclhva2Qm(gq4Bz9sJGyLDzVHGWjAWJj4zG3KP4lf2knnxTt1CEsX6S8LkM30BIgPtyu8Spbge4da3XFb3ZkgkxcpEhqTZKPkWZo4DV37Ei5HbbqHglslDCGOZ0Ck2c3EsCg3AJMrMBmsCzoefR0l)P7t1kFqSprG8SYKI6r0MueTjfrBsr0gN8kKzUsUK7e4l7ZdgO85jG4J5zEB2FcYdxfVqB6BaNg9b8Qx1)kWTdzbPW9ygYa9XZKnGZvgNA05RQecd1zmfkJ8xbNxjZl1smSeo6Ykw7LKIY)75CJOyA)8SSIPxVqcklZOxVlIedouhx56dzLGbZg3(TdUIHm54okeTTEPQk4mm7YbV6OvUDXdNo(UQO6BACkwPiH6VFjui4ftFzX0RgumT39oHc5XcLAKrthDzUjLpldu0oqMjCwlFz21YeSM9vqnb6VGuFUJtQDnxP3DX6nSx3W(JBFrpW5lYMx6s6mkmC0fKDoxM6v1CuXUiy9534Ei8QVVgFmn899Jgp4ZJccBdj6wTbdW2yHkrQsjbKiRzVgEIm3crd0s9jcOTxROUSsuPvWxb3I6FfkeeFHcu8gt8Czf4dy9mgU7IWlTKlv)ef3w4LFsc0xJTfE6GaNn2lrTBVGWEJrNQolrVwXwlbj2zhbk9D2ptSs7c96UzGkBX35rGKieIhkTlbv05)aPec1XlewgiXrkVKhPAdA6DINbH0FTjXAaNC0kb4DX6T)EFmApmCfIqT7eXACuy56ApPGPYyWmChUfXbWQzjs7xZvL2axTbcf5wNEzONTHAuIS5elA1VvE7xu8HIP(FmcxUrvmT8K)P1xkMQn7238lfFaQL8H07eLY9cDcGd9YMn8H1d0D4BcXQ2iUUBmMpsnjoy1wmIypEh479EGISvl4EZ3cWhY0fUsw7uI)blF3ow7dwyeRYO2AVRm6uJWA)ZIP0dc)nrK541Ca1Dcepft)JIPBVzX0F)P3bCi(n(Kqqbv2SYokibsLWn7xQwRcKT9w)G6(nvlUTYkUPAXTpN9CZXmOBEXwjEmwU9WgnfO2SZ77rnOV3tFRqHIy1SZhhod3602de7SYUycssuGUn9uhbA7)dqTq9SL7q(PCKhnuujKN4E3e7coGIEtn4xIm(jI9R2hX1pWSREUFAtvIomlGglhJ9DRzTH9LhbquIQM4NFwvpGB3xT(kG0iNVzxBbFylK02Grb9Q8FG56nb6AXp7iA7u85AuN7BKpum9J58eCicy8Jh7REIsTgM55AdhM966TROdUtALWSkuxeY(R3yBqqqVXt8(NYHy3Qd(P3A2QYp8KWdYKpEp6QWHda72HMfiJ0m96(gX3ZfQyS84KwVYtSeO4DzvXXzE6EXml350QrWCzWu9(dQ()GA2WVVUDNM94SHJ5uRIfKcq36M9Ks4cz83GwTwQEdPW2opmPfsSpCWB876unHmLHVqqG0bTEZUH8pYi8hAAFe2ub9ijxXrgYV(C2YyeD5bTEzTXuQpyz3X96fuYxTjuWSZ3kT4yRj0KkyAFNJoh8HgXTwfvvMFjki4Mw7lBo98sORNpMqJ0MXB(Xx39ZFk)JlXMPOAqOe1N99atUV)KTvwiAn7n3n59UdwDwlCt0)o

**Comment by mrbuds** [comment]
> Which version of WeakAuras are you using?
> 2.17.4

Current version of WeakAuras is 2.17.9, can you replicate the issue using last version ?

**Comment by Weeeqt** [comment]
> Current version of WeakAuras is 2.17.9, can you replicate the issue using last version ?

Yes, just updated and can confirm that issue still exists.

I also found out that if I add "Remaining Time > 0" to Aura trigger then the issue seems to be gone. I created two auras and placed them side by side for comparison. The one without "remaining time" still has the bug.

I suppose it's not a big deal if it works as expected with "Remaining Time > 0", but I guess it's still a minor bug which may occasionally happen (because of the timing it requires).
